modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -14,9 +14,76 @@ const defaultProps = {
   highlightedFeatureId: null
 };
 
+async function fetchTileJSON(url) {
+  try {
+    return await load(url);
+  } catch (error) {
+    throw new Error(`An error occurred fetching TileJSON: ${error}`);
+  }
+}
+
 export default class MVTLayer extends TileLayer {
+  initializeState() {
+    super.initializeState();
+    this.setState({
+      data: null,
+      tileJSON: null
+    });
+  }
+
+  get isLoaded() {
+    return this.state.data && this.state.tileset && super.isLoaded;
+  }
+
+  updateState({props, oldProps, context, changeFlags}) {
+    if (changeFlags.dataChanged) {
+      this._updateTileData({props});
+    }
+
+    if (this.state.data) {
+      super.updateState({props, oldProps, context, changeFlags});
+    }
+  }
+
+  async _updateTileData({props}) {
+    const {onDataLoad} = this.props;
+    let {data} = props;
+    let tileJSON = null;
+    let {minZoom, maxZoom} = props;
+
+    if (typeof data === 'string' && !isURLTemplate(data)) {
+      this.setState({data: null, tileJSON: null});
+      tileJSON = await fetchTileJSON(data);
+
+      if (tileJSON.minzoom && tileJSON.minzoom > minZoom) {
+        minZoom = tileJSON.minzoom;
+      }
+
+      if (tileJSON.maxzoom && (!maxZoom || tileJSON.maxzoom < maxZoom)) {
```

Comment:
`maxZoom` can be 0, use `Number.isFinite`.

Reply:
Done!